lib/elixir/pages/Typespecs.md

```diff
@@ -25,65 +25,72 @@ The syntax Elixir provides for type specifications is similar to [the one in Erl
     type :: any()                   # the top type, the set of all terms
           | none()                  # the bottom type, contains no terms
           | atom()
-          | float()
-          | integer()
-          | neg_integer()           # ..., -3, -2, -1
-          | non_neg_integer()       # 0, 1, 2, 3, ...
-          | pos_integer()           # 1, 2, 3, ...
-          | list(type)
-          | nonempty_list(type)
-          | improper_list(type1, type2)
-          | maybe_improper_list(type1, type2)
           | map()                   # any map
           | pid()
           | port()
           | reference()
           | struct()                # any struct
-          | tuple()
+          | tuple()                 # tuple of any size
+
+                                    ## Numbers
+          | float()                 # float
+          | integer()               # integer
+          | neg_integer()           # ..., -3, -2, -1
+          | non_neg_integer()       # 0, 1, 2, 3, ...
+          | pos_integer()           # 1, 2, 3, ...
+
+                                                        ## Lists
+          | list(type)                                  # proper list ([]-terminated)
+          | nonempty_list(type)                         # non-empty proper list
+          | maybe_improper_list(type1, type2)           # proper or improper list
+          | nonempty_improper_list(type1, type2)        # improper list
+          | nonempty_maybe_improper_list(type1, type2)  # non-empty proper or improper list
+
           | Literals                # Described in section "Literals"
-          | Builtin                 # Described in section "Built-in types"
+          | BuiltIn                 # Described in section "Built-in types"
```

Comment:
I tried that, but it was highlighted weirdly (until the hyphen only). Then i thought its not a valid atom. I dont care about the highlighting though. Let me know what u think